src/main/java/client/ClientTemp.java

```diff
@@ -75,7 +75,7 @@ public static void main(String[] args) throws Exception {
                 Message generatedMessage = clientHandler.generateMessage(user, message);
                 session.writeMessage(generatedMessage);
             } catch (Exception e) {
-                if (e instanceof IndexOutOfBoundsException) {
+                if (e instanceof IllegalArgumentException) {
                     logger.error("Failed to send message", e);
                     continue;
                 }
```

Comment:
When you use the common exception as a marker that something went wrong but it is ok, there is a chance that this exception will be thrown by something else that you don't expect.
Because of that, it is always better to have your own exception for that purpose.
Also, Java allows you to define multiple catch blocks for different types of exceptions. It can be used instead of checking the exception with `instanceof` manually